### PR TITLE
Compress responses

### PR DESCRIPTION
**Pull request by andycroll**

### CODE REVIEW

File: _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+category: ruby
+image:
+  base: '2021/compress-your-rails-html-responses-on-heroku'
+  alt: "Folded paper texture"
+  credit: "JJ Ying"
+  source: "https://unsplash.com/photos/WmnsGyaFnCQ"
+
+---
+
+There are multiple aspects to the performance of a typical Rails applications: database queries, view render time, fragment caching and more. There are also considerations for the front-end: image compression, Javascript & CSS size.
```
Comment (nodunayo) [inline _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md]: *Oxford comma, pretty please!

*,and CSS size.

File: _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+We often use CDNs, compression and minification for the assets (images, CSS & JS) our applications use. However, many hosts ([including Heroku](https://devcenter.heroku.com/articles/compressing-http-messages-with-gzip)) do not compress the HTML that is generated by our applications by default.
```
Comment (nodunayo) [inline _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md]: *Oxford comma, pretty please!

Comment (nodunayo) [reply _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md]: What is it with these ampersands, eh?!

File: _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  end
+end
+```
+
+
+## Why?
+
+Compressing 'text-based' resources (i.e. the HTML or JSON your views generate) is a [standard performance recommendation from Google](https://web.dev/uses-text-compression/).
+
+In most cases text-based responses compress _really_ well. I’ve seen compression of 20x on some of our more “HTML-repetative” `#index` actions at work, but 4x is typical on smaller pages. It makes an even bigger differenceto performance on mobile, where the network is more of a bottleneck.
```
Comment (nodunayo) [inline _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md]:
```suggestion
In most cases, text-based responses compress _really_ well. I’ve seen compression of 20x on some of our more “HTML-repetative” `#index` actions at work, but 4x is typical on smaller pages. It makes an even bigger difference to performance on mobile, where network is more of a bottleneck.
```

File: _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+You might find _spectacular_ savings depending on your HTML output if you're using [TailwindCSS](https://tailwindcss.com), [Bootstrap](https://getbootstrap.com) or another framework that result in repetative class names in the HTML.
```
Comment (nodunayo) [inline _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md]:
```suggestion
You might find _spectacular_ savings if you're using something like [TailwindCSS](https://tailwindcss.com), [Bootstrap](https://getbootstrap.com), or another framework that result in repetitive class names in the HTML.
```

File: _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+This (sadly!) doesn't absolve you of other performance work. If your database queries are slow or your views take a long time to generate you’ll still need to address that, as these compression savings occur after the page has rendered.
```
Comment (nodunayo) [inline _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md]: *comma after generate

*...savings occur only after the page has rendered.

**Review by nodunayo — DISMISSED** [review]

File: _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md (hunk 1 above)
Comment (andycroll) [reply _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md]:
```suggestion
There are multiple aspects to the performance of a typical Rails applications: database queries, view render time, fragment caching, and more. There are also considerations for the front-end: image compression, Javascript, and CSS size.
```

File: _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md (hunk 2 above)
Comment (andycroll) [reply _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md]:
```suggestion
We often use CDNs, compression, and minification for the assets (images, CSS, and JS) our applications use. However, many hosts ([including Heroku](https://devcenter.heroku.com/articles/compressing-http-messages-with-gzip)) do not compress the HTML that is generated by our applications by default.
```

File: _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md (hunk 5 above)
Comment (andycroll) [reply _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md]:
```suggestion
This (sadly!) doesn't absolve you of other performance work. These compression savings only occur after the page has rendered, so if your database queries are slow or your views take a long time to generate, you’ll still need to address those issues.
```

File: _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+category: ruby
+image:
+  base: '2021/compress-your-rails-html-responses-on-heroku'
+  alt: "Folded paper texture"
+  credit: "JJ Ying"
+  source: "https://unsplash.com/photos/WmnsGyaFnCQ"
+
+---
+
+There are multiple aspects to the performance of a typical Rails applications: database queries, view render time, fragment caching, and more. There are also considerations for the front-end: image compression, Javascript, and CSS size.
```
Comment (nodunayo) [inline _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md]: *of typical

File: _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    config.middleware.use Rack::Deflater
+    # ...
+  end
+end
+```
+
+
+## Why?
+
+Compressing 'text-based' resources (i.e. the HTML or JSON your views generate) is a [standard performance recommendation from Google](https://web.dev/uses-text-compression/).
```
Comment (nodunayo) [inline _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md]: 'from' or 'by' Google? 🤔

File: _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+In most cases, text-based responses compress _really_ well. I’ve seen compression of 20x on some of our more “HTML-repetative” `#index` actions at work, but 4x is typical on smaller pages. It makes an even bigger difference to performance on mobile, where network is more of a bottleneck.
+
+You might find _spectacular_ savings if you're using something like [TailwindCSS](https://tailwindcss.com), [Bootstrap](https://getbootstrap.com), or another framework that result in repetitive class names in the HTML.
```
Comment (nodunayo) [inline _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md]: *that results

File: _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+This (sadly!) doesn't absolve you of other performance work. These compression savings only occur after the page has rendered, so if your database queries are slow or your views take a long time to generate, you’ll still need to address those issues.
```
Comment (nodunayo) [inline _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md]: Comma before "or your views take a long time..."

File: _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+
+## Why not?
+
+It is entirely possible you already serve your application from behind a CDN, like Fastly or Cloudflare, that automatically compresses your HTML & JSON responses. You won't need `Rack::Deflater` in these cases.
```
Comment (nodunayo) [inline _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md]: *and (not &)

File: _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+
+There will be a small increase in processing time for your user's devices as they decompress the response, but this is _vastly_ outweighed by the time savings made from sending the compressed response over the Internet.
```
Comment (nodunayo) [inline _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md]: *users'

File: _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md (hunk 6 above)
Comment (andycroll) [reply _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md]:
```suggestion
There are multiple aspects to the performance of typical Rails applications: database queries, view render time, fragment caching, and more. There are also considerations for the front-end: image compression, Javascript, and CSS size.
```

File: _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md (hunk 7 above)
Comment (andycroll) [reply _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md]:
```suggestion
Compressing 'text-based' resources (i.e. the HTML or JSON your views generate) is one of Google‘s [standard performance recommendations](https://web.dev/uses-text-compression/).
```

File: _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md (hunk 8 above)
Comment (andycroll) [reply _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md]:
```suggestion
You might find _spectacular_ savings if you're using something like [TailwindCSS](https://tailwindcss.com), [Bootstrap](https://getbootstrap.com), or another framework that results in repetitive class names in the HTML.
```

File: _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md (hunk 10 above)
Comment (andycroll) [reply _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md]:
```suggestion
It is entirely possible you already serve your application from behind a CDN, like Fastly or Cloudflare, that automatically compresses your HTML and JSON responses. You won't need `Rack::Deflater` in these cases.
```

File: _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md (hunk 9 above)
Comment (andycroll) [reply _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md]:
```suggestion
This (sadly!) doesn't absolve you of other performance work. These compression savings only occur after the page has rendered, so if your database queries are slow, or your views take a long time to generate, you’ll still need to address those issues.
```

File: _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md (hunk 11 above)
Comment (andycroll) [reply _posts/2021/2021-08-30-compress-your-rails-html-responses-on-heroku.md]:
```suggestion
There will be a small increase in processing time for your users’ devices as they decompress the response, but this is _vastly_ outweighed by the time savings made from sending the compressed response over the Internet.
```

**Pull request merged**